### DISCUSSION TITLE
What is this esoteric translation of REST queries into SQL?

### DISCUSSION THREAD

**Discussion by mspi21** [directus/directus#15586] [Archive]
I'm sure that Directus' "Global Query" language is capable of creating _any_ basic query representable in SQL, such as selecting the top 10 highest value rows in a grouped aggregation.

However, so far, I've mostly been unsuccessful in trying to accomplish creating such a basic query and the documentation did not make it any clearer.

Here are two examples of very similar (almost identical!) queries in SQL:

```sql
select user_created, count(*) from users_posts
group by user_created
order by count(*) desc
limit 10;
```

```sql
select user_created, avg(rating) from users_ratings
group by user_created
order by avg(rating) desc
limit 10;
```

The way I tried composing these using Directus' REST API is the following:
```url
GET /items/users_posts/?aggregate[count]=id&groupBy[]=user_created&sort=-count&limit=10
GET /items/users_ratings/?aggregate[avg]=rating&groupBy[]=user_created&sort=-avg&limit=10
```

The result is suprising, in that the first query is valid and successful, whereas the latter produces the error `column users_ratings.avg does not exist`. Why? Aren't `count` and `avg` perfectly equivalent aggregate functions, used in the exact same way? Why are they producing different behaviour? How can I implement this query using "Global Parameters"?

**Comment by licitdev** [comment]
```
GET /items/users_ratings/?aggregate[avg]=rating&groupBy[]=user_created&sort=-rating&limit=10
```
Please give the above query a try. I believe sort should be the column name, which would be `rating` in your example. 🤔
